content/en/docs/reference/labels-annotations-taints/_index.md

```diff
@@ -264,6 +264,19 @@ in the StatefulSet topic for more details.
 Note the [PodIndexLabel](/docs/reference/command-line-tools-reference/feature-gates/)
 feature gate must be enabled for this label to be added to pods.
 
+### resource.kubernetes.io/pod-claim-name
+
+Type: Annotation
+
+Example: `resource.kubernetes.io/pod-claim-name: "my-pod-claim"`
+
+Used on: ResourceClaim
+
+This annotation is assigned to generated ResourceClaims. 
+Its value corresponds to the name of the resource claim in the `.spec` of any Pod(s) for which the ResourceClaim was created.
+This annotation is an internal implementation detail of [dynamic resource allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/).
+You should not need to read or modify the value of this annotation.
+
```

Comment:
It's mentioned [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/resourceclaim/controller.go#L58-L59) that this annotation `is used only inside the controller and not documented as part of the Kubernetes API.`
It looks like it's intentionally not documented. @pohly is it?

Reply:
This is indeed intentional.

Reply:
OK, **but** per https://github.com/kubernetes/website/issues/47044 we don't have a way to register annotations without also documenting them.

Reply:
(adding such a mechanism is welcome, and it's not actually a huge amount of work, but it'd be bigger than this PR)

Reply:
/lgtm